measures/options.txt
`configure' configures XZ Utils 5.3.1alpha to adapt to many kinds of systems.

Usage: ./configure [OPTION]... [VAR=VALUE]...

To assign environment variables (e.g., CC, CFLAGS...), specify them as
VAR=VALUE.  See below for descriptions of some of the useful variables.

Defaults for the options are specified in brackets.

Configuration:
  -h, --help              display this help and exit
      --help=short        display options specific to this package
      --help=recursive    display the short help of all the included packages
  -V, --version           display version information and exit
  -q, --quiet, --silent   do not print `checking ...' messages
      --cache-file=FILE   cache test results in FILE [disabled]
  -C, --config-cache      alias for `--cache-file=config.cache'
  -n, --no-create         do not create output files
      --srcdir=DIR        find the sources in DIR [configure dir or `..']

Installation directories:
  --prefix=PREFIX         install architecture-independent files in PREFIX
                          [/usr/local]
  --exec-prefix=EPREFIX   install architecture-dependent files in EPREFIX
                          [PREFIX]

By default, `make install' will install all the files in
`/usr/local/bin', `/usr/local/lib' etc.  You can specify
an installation prefix other than `/usr/local' using `--prefix',
for instance `--prefix=$HOME'.

For better control, use the options below.

Fine tuning of the installation directories:
  --bindir=DIR            user executables [EPREFIX/bin]
  --sbindir=DIR           system admin executables [EPREFIX/sbin]
  --libexecdir=DIR        program executables [EPREFIX/libexec]
  --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
  --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
  --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
  --libdir=DIR            object code libraries [EPREFIX/lib]
  --includedir=DIR        C header files [PREFIX/include]
  --oldincludedir=DIR     C header files for non-gcc [/usr/include]
  --datarootdir=DIR       read-only arch.-independent data root [PREFIX/share]
  --datadir=DIR           read-only architecture-independent data [DATAROOTDIR]
  --infodir=DIR           info documentation [DATAROOTDIR/info]
  --localedir=DIR         locale-dependent data [DATAROOTDIR/locale]
  --mandir=DIR            man documentation [DATAROOTDIR/man]
  --docdir=DIR            documentation root [DATAROOTDIR/doc/xz]
  --htmldir=DIR           html documentation [DOCDIR]
  --dvidir=DIR            dvi documentation [DOCDIR]
  --pdfdir=DIR            pdf documentation [DOCDIR]
  --psdir=DIR             ps documentation [DOCDIR]

Program names:
  --program-prefix=PREFIX            prepend PREFIX to installed program names
  --program-suffix=SUFFIX            append SUFFIX to installed program names
  --program-transform-name=PROGRAM   run sed PROGRAM on installed program names

System types:
  --build=BUILD     configure for building on BUILD [guessed]
  --host=HOST       cross-compile to build programs to run on HOST [BUILD]

Optional Features:
  --disable-option-checking  ignore unrecognized --enable/--with options
  --disable-FEATURE       do not include FEATURE (same as --enable-FEATURE=no)
  --enable-FEATURE[=ARG]  include FEATURE [ARG=yes]
  --enable-debug          Enable debugging code.
  --enable-encoders=LIST  Comma-separated list of encoders to build.
                          Default=all. Available encoders: lzma1 lzma2 delta
                          x86 powerpc ia64 arm armthumb sparc
  --enable-decoders=LIST  Comma-separated list of decoders to build.
                          Default=all. Available decoders are the same as
                          available encoders.
  --enable-match-finders=LIST
                          Comma-separated list of match finders to build.
                          Default=all. At least one match finder is required
                          for encoding with the LZMA1 and LZMA2 filters.
                          Available match finders: hc3 hc4 bt2 bt3 bt4
  --enable-checks=LIST    Comma-separated list of integrity checks to build.
                          Default=all. Available integrity checks: crc32 crc64
                          sha256
  --enable-external-sha256
                          Use SHA-256 code from the operating system. See
                          INSTALL for possible subtle problems.
  --disable-assembler     Do not use assembler optimizations even if such
                          exist for the architecture.
  --enable-small          Make liblzma smaller and a little slower. This is
                          disabled by default to optimize for speed.
  --enable-threads=METHOD Supported METHODS are `yes', `no', `posix', `win95',
                          and `vista'. The default is `yes'. Using `no'
                          together with --enable-small makes liblzma thread
                          unsafe.  ###################################################### !!!!
  --enable-assume-ram=SIZE
                          If and only if the real amount of RAM cannot be
                          determined, assume SIZE MiB. The default is 128 MiB.
                          This affects the default memory usage limit.
  --disable-xz            do not build the xz tool
  --disable-xzdec         do not build xzdec
  --disable-lzmadec       do not build lzmadec (it exists primarily for LZMA
                          Utils compatibility)
  --disable-lzmainfo      do not build lzmainfo (it exists primarily for LZMA
                          Utils compatibility)
  --disable-lzma-links    do not create symlinks for LZMA Utils compatibility
  --disable-scripts       do not install the scripts xzdiff, xzgrep, xzless,
                          xzmore, and their symlinks
  --disable-doc           do not install documentation files to docdir (man
                          pages will still be installed)
  --enable-symbol-versions
                          Use symbol versioning for liblzma. Enabled by
                          default on GNU/Linux, other GNU-based systems, and
                          FreeBSD.
  --enable-sandbox=METHOD Sandboxing METHOD can be `auto', `no', or
                          `capsicum'. The default is `auto' which enables
                          sandboxing if a supported sandboxing method is
                          found.
  --enable-path-for-scripts=PREFIX
                          If PREFIX isn't empty, PATH=PREFIX:$PATH will be set
                          in the beginning of the scripts (xzgrep and others).
                          The default is empty except on Solaris the default
                          is /usr/xpg4/bin.
  --enable-silent-rules   less verbose build output (undo: "make V=1")
  --disable-silent-rules  verbose build output (undo: "make V=0")
  --enable-dependency-tracking
                          do not reject slow dependency extractors
  --disable-dependency-tracking
                          speeds up one-time build
  --enable-shared[=PKGS]  build shared libraries [default=yes]
  --enable-static[=PKGS]  build static libraries [default=yes]
  --enable-fast-install[=PKGS]
                          optimize for fast installation [default=yes]
  --disable-libtool-lock  avoid locking (might break parallel builds)
  --disable-nls           do not use Native Language Support
  --disable-rpath         do not hardcode runtime library paths
  --disable-largefile     omit support for large files
  --enable-unaligned-access
                          Enable if the system supports *fast* unaligned
                          memory access with 16-bit and 32-bit integers. By
                          default, this is enabled only on x86, x86_64, big
                          endian PowerPC, and some ARM systems.
  --enable-unsafe-type-punning
                          This introduces strict aliasing violations and may
                          result in broken code. However, this might improve
                          performance in some cases, especially with old
                          compilers (e.g. GCC 3 and early 4.x on x86, GCC < 6
                          on ARMv6 and ARMv7).
  --enable-werror         Enable -Werror to abort compilation on all compiler
                          warnings.

Optional Packages:
  --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
  --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
  --with-pic[=PKGS]       try to use only PIC/non-PIC objects [default=use
                          both]
  --with-aix-soname=aix|svr4|both
                          shared library versioning (aka "SONAME") variant to
                          provide on AIX, [default=aix].
  --with-gnu-ld           assume the C compiler uses GNU ld [default=no]
  --with-sysroot[=DIR]    Search for dependent libraries within DIR (or the
                          compiler's sysroot if not specified).
  --with-gnu-ld           assume the C compiler uses GNU ld [default=no]
  --with-libiconv-prefix[=DIR]  search for libiconv in DIR/include and DIR/lib
  --without-libiconv-prefix     don't search for libiconv in includedir and libdir
  --with-libintl-prefix[=DIR]  search for libintl in DIR/include and DIR/lib
  --without-libintl-prefix     don't search for libintl in includedir and libdir

Some influential environment variables:
  CC          C compiler command
  CFLAGS      C compiler flags
  LDFLAGS     linker flags, e.g. -L<lib dir> if you have libraries in a
              nonstandard directory <lib dir>
  LIBS        libraries to pass to the linker, e.g. -l<library>
  CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
              you have headers in a nonstandard directory <include dir>
  CCAS        assembler compiler command (defaults to CC)
  CCASFLAGS   assembler compiler flags (defaults to CFLAGS)
  CPP         C preprocessor
  LT_SYS_LIBRARY_PATH
              User-defined run-time library search path.

Use these variables to override the choices made by `configure' or to help
it to find libraries and programs with nonstandard names/locations.

Report bugs to <lasse.collin@tukaani.org>.
XZ Utils home page: <https://tukaani.org/xz/>.


measures/options.py
import os, glob

# The following 8 arrays contain the extracted compile-time 
# configuration options from the xz using the "./configure --help"
optional_features = ["--disable%s" % p for p in ("-option-checking", 
                                                "-assembler",
                                                #"-xz",
                                                "-xzdec",
                                                "-lzmadec",
                                                "-lzmainfo",
                                                "-lzma-links",
                                                "-scripts",
                                                "-doc",
                                                "-silent-rules",
                                                "-dependency-tracking",
                                                "-libtool-lock",
                                                "-nls",
                                                "-rpath",
                                                "-largefile"
                                                )]              
optional_features_2 = ["--enable%s" % p for p in ("-debug",
                                                "-encoders=lzma1",
                                                "-encoders=lzma1,lzma2", # configure: error: LZMA2 requires that LZMA1 is also enabled.
                                                "-encoders=lzma1,delta", # an error !
                                                "-encoders=lzma1,x86",
                                                "-encoders=lzma1,powerpc",
                                                "-encoders=lzma1,ia64",
                                                "-encoders=lzma1,arm",
                                                "-encoders=lzma1,armthumb",
                                                "-encoders=lzma1,sparc",
                                                "-decoders=lzma1",
                                                "-decoders=lzma1,lzma2",
                                                "-decoders=lzma1,delta",
                                                "-decoders=lzma1,x86",
                                                "-decoders=lzma1,powerpc",
                                                "-decoders=lzma1,ia64",
                                                "-decoders=lzma1,arm",
                                                "-decoders=lzma1,armthumb",
                                                "-decoders=lzma1,sparc",
                                                "-match-finders=hc3",
                                                "-match-finders=hc4",
                                                "-match-finders=bt2",
                                                "-match-finders=bt3",
                                                "-match-finders=bt4",
                                                "-checks=crc32",
                                                "-checks=crc32,crc64", # configure: error: For now, the CRC32 check must always be enabled.
                                                "-checks=crc32,sha256", #configure: error: For now, the CRC32 check must always be enabled.
                                                #"-external-sha256", # configure: error: --enable-external-sha256 was specified but no supported external SHA-256 implementation was found
                                                "-small",
                                                "-threads=yes",
                                                "-threads=no",
                                                "-threads=posix",
                                                #"-threads=win95", # error
                                                #"-threads=vista", # error
                                                "-assume-ram=128",
                                                "-symbol-versions",
                                                "-sandbox=auto",
                                                "-sandbox=no",
                                                "-sandbox=capsicum",
                                                "-silent-rules",
                                                "-dependency-tracking",
                                                #"-shared",
                                                "-shared=no",
                                                "-static",
                                                #"-static=no",
                                                "-fast-install",
                                                "-fast-install=no",
                                                "-unaligned-access",
                                                "-unsafe-type-punning",
                                                #"-werror"
                                                ) ]

optional_packages = ["--with%s" % p for p in ("-pic=PIC",
                                            "-pic=non-PIC",
                                            "-aix-soname=aix",
                                            "-aix-soname=svr4",
                                            "-aix-soname=both",
                                            "-gnu-ld",
                                            "-gnu-ld=no",
                                            "-libiconv-prefix",
                                            "-libintl-prefix"
                                            ) ]


optional_packages_2 = ["--without%s" % p for p in ( "-libiconv-prefix",
                                                    "-libintl-prefix"
                                            ) ]



# Adding the single configurations to an array 
def extractDigits(lst):
    res = []
    for el in lst:
        sub = el.split(', ')
        res.append(sub)
    #print(res)
    return(res)
   

# All single configurations                   
single_configurations_01 = extractDigits(optional_features)
single_configurations_02 = extractDigits(optional_features_2)
single_configurations_03 = extractDigits(optional_packages)
single_configurations_04 = extractDigits(optional_packages_2)


# Adding the sample configuration sets (generated by FeatureIDE) to an array
sample_configurations = []

# for variant in glob.glob("measures/products/*.config"):
for variant in glob.glob("measures/products_2-200/*.config"):
    lineList = list()
    with open(variant) as f:
        for line in f:
            lineList = [line.rstrip('\n') for line in open(variant)]
        sample_configurations.append(lineList)

# print(sample_configurations)
print(*single_configurations_01,*single_configurations_02,*single_configurations_03,*single_configurations_04)

# All single and sample configurations, 
# which will be used to measure the changes on
# the binary size and number of gadgets in x264
all_options =  [*sample_configurations]